Tests/ML/test_model_train_test_and_recovery.py

```diff
@@ -0,0 +1,90 @@
+from InnerEye.ML.configs.classification.DummyClassification import DummyClassification
+from InnerEye.ML.model_training import model_train
+from InnerEye.ML.model_testing import model_test
+from InnerEye.ML.utils.run_recovery import RunRecovery
+from InnerEye.ML.metrics import InferenceMetricsForClassification
+
+from Tests.ML.util import get_default_checkpoint_handler
+
+
+@pytest.mark.parametrize("mean_teacher_model", [True, False])
+def test_recover_testing_from_run_recovery(mean_teacher_model: bool,
+                                            test_output_dirs: TestOutputDirectories) -> None:
+    """
+    Checks that inference results are the same whether from a checkpoint in the same run, from a run recovery or from a
+    local_weights_path param.
+    """
+    # Train for 4 epochs
+    config = DummyClassification()
+    if mean_teacher_model:
+        config.mean_teacher_alpha = 0.999
+    config.set_output_to(Path(test_output_dirs.root_dir) / "original")
+    os.makedirs(str(config.outputs_folder))
+    config.save_start_epoch = 2
+    config.save_step_epochs = 2
+
+    checkpoint_handler = get_default_checkpoint_handler(model_config=config,
+                                                        project_root=Path(test_output_dirs.root_dir))
+    train_results = model_train(config, checkpoint_handler=checkpoint_handler)
+    assert len(train_results.learning_rates_per_epoch) == 4
+
+    # Run inference on this
+    checkpoint_handler.additional_training_done()
+    test_results = model_test(config=config, data_split=ModelExecutionMode.TEST, checkpoint_handler=checkpoint_handler)
+    assert isinstance(test_results, InferenceMetricsForClassification)
+    assert list(test_results.epochs.keys()) == [4]
+
+    # Mimic using a run recovery and see if it is the same
+    config_run_recovery = DummyClassification()
+    if mean_teacher_model:
+        config_run_recovery.mean_teacher_alpha = 0.999
+    config_run_recovery.set_output_to(Path(test_output_dirs.root_dir) / "run_recovery")
+    os.makedirs(str(config_run_recovery.outputs_folder))
+
+    checkpoint_handler_run_recovery = get_default_checkpoint_handler(model_config=config_run_recovery,
+                                                                     project_root=Path(test_output_dirs.root_dir))
+    # make if seem like run recovery objects have been downloaded
+    checkpoint_root = Path(config_run_recovery.checkpoint_folder) / "recovered"
+    shutil.copytree(config.checkpoint_folder, checkpoint_root)
+    checkpoint_handler_run_recovery.run_recovery = RunRecovery([checkpoint_root])
+    test_results_run_recovery = model_test(config_run_recovery, data_split=ModelExecutionMode.TEST,
+                                           checkpoint_handler=checkpoint_handler_run_recovery)
+    assert isinstance(test_results_run_recovery, InferenceMetricsForClassification)
+    assert list(test_results_run_recovery.epochs.keys()) == [4]
```

Comment:
here and below: replace 4 with whatever it is coming from (config.num_epochs probably)